Firestore/core/test/firebase/firestore/auth/user_test.cc

```diff
@@ -49,6 +49,12 @@ TEST(User, Comparison) {
   EXPECT_NE(User("abc"), User("xyz"));
 }
 
+TEST(User, Hash) {
+  const HashUser hash;
+  EXPECT_NE(hash(User("abc")), hash(User("xyz")));
+  EXPECT_NE(hash(User()), hash(User("abc")));
+}
```

Comment:
Since you're testing the hash, you may was well through in an equality check too.

EXPECT_EQ(hash(User("abc")), hash(User("abc"));

Reply:
These assertions are not guaranteed to be true. A valid implementation of std::hash is `return 42;` and this would fail that implementation.

All you can assert is that for two values that are equal the hashes should be equal. The inverse is not necessarily true.

Reply:
done